src/librustc/ty/subst.rs

```diff
@@ -70,6 +71,28 @@ impl<'tcx> UnpackedKind<'tcx> {
     }
 }
 
+impl<'tcx> Ord for Kind<'tcx> {
+    fn cmp(&self, other: &Kind) -> Ordering {
+        match (self.unpack(), other.unpack()) {
+            (UnpackedKind::Type(_), UnpackedKind::Lifetime(_)) => Ordering::Greater,
```

Comment:
You don't need to implement all of this manually, just add `#[derive(PartialEq, Eq, PartialOrd, Ord)]` on `UnpackedKind` and implement this method going through that. That's how other traits are implemented on `Kind`.

Reply:
oops missed this one. Here is a pr for that https://github.com/rust-lang/rust/pull/54074